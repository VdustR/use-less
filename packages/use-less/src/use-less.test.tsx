import React, { useState, FC } from 'react';
import { render, fireEvent } from '@testing-library/react';
import useLess from './use-less';

const TestComponent: FC = () => {
  const [count, setCounter] = useState(0);
  useLess();

  return (
    <button onClick={() => setCounter((count) => count + 1)}>
      Click to increase: {count}
    </button>
  );
};

it('will fail every time', () => {
  const { getByText, asFragment } = render(<TestComponent />);
  const firstRender = asFragment();
  expect(firstRender).toMatchSnapshot();

  fireEvent.click(getByText(/Click to increase/));
  const secondRender = asFragment();
  expect(secondRender).toMatchSnapshot();
});
