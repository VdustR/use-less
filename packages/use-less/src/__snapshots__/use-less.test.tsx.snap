// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will fail every time 1`] = `
<DocumentFragment>
  <button>
    Click to increase: 0
  </button>
</DocumentFragment>
`;

exports[`will fail every time 2`] = `
<DocumentFragment>
  <button>
    Click to increase: 1
  </button>
</DocumentFragment>
`;
